Source/WebInspectorUI/UserInterface/Views/ComputedStyleDetailsPanel.js

```diff
@@ -89,47 +89,43 @@ WI.ComputedStyleDetailsPanel = class ComputedStyleDetailsPanel extends WI.StyleD
 
     refresh(significantChange)
     {
-        // We only need to do a rebuild on significant changes. Other changes are handled
-        // by the sections and text editors themselves.
+        // We only need to do a rebuild on significant changes.
+        // Sections update their contents in response to changes to `DOMNodesStyles.computedStyle`.
+        // Sections for variable groups are added/removed based on available data to avoid needless updates.
         if (!significantChange) {
-            super.refresh();
-            return;
-        }
-
-        this._computedStyleSection.styleTraces = this._computePropertyTraces(this.nodeStyles.uniqueOrderedStyles);
-        this._computedStyleSection.style = this.nodeStyles.computedStyle;
-        this._propertiesSection.element.classList.toggle("hidden", !this._computedStyleSection.propertiesToRender.length);
-        this._boxModelDiagramRow.nodeStyles = this.nodeStyles;
-
-        let styleGroups = new Map;
-
-        switch (this.variablesGroupingMode) {
-        case WI.ComputedStyleDetailsPanel.VariablesGroupingMode.Ungrouped:
-            styleGroups.set(this._variablesStyleSectionForGroupTypeMap.get(WI.DOMNodeStyles.VariablesGroupType.Ungrouped), this.nodeStyles.computedStyle);
-            break;
+            super.refresh(significantChange);
```

Comment:
it seems like this is called in both paths.  should we maybe just move it to the top of this function and only call it once instead?